### ISSUE TITLE
[a11y]: Select component implementation issues 

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#13008]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-select--default#storybook-preview-wrapper

### Automated testing tool and ruleset

n/a

### Description

During a rapid review of Carbon components, we identified a number of accessibility considerations. This issue captures those concerns, along with some implementation challenges identified by our more detailed review of the Select component, as part of our update of the accessibility content. 
Findings that are not specifically implementation issues are captured in a [separate issue on the carbon website repo](https://github.com/carbon-design-system/carbon-website/issues/3361). For the curious, the [original recording of the quick review](https://ibm.webex.com/ibm/ldr.php?RCID=210905c96aad3c6f890cc94fdc4f9523) is reachable (Password: wHhv2TW9 ). Select begins at about 11:25 in the recording.

Recording link: 

Key findings:

1. The default text "Choose an option" is not an actual option in the list. Effectively Carbon is using placeholder text to provide instructions. This is non-optimal for a number of reasons, gone into in the usage.pdf comments. Suffice it to say, that typically, the pre-populated option in a select should be one of the options in the list. It is _okay_ to have a blank, and that would be the recommendation if you do not want to have a  pre-populated option selected. Having a blank also arguably makes it easier for a user to tell when a select has not been completed.
The third item in https://github.com/carbon-design-system/carbon/issues/12721 about dropdowns covers the same basic problem.
I am aware that it is _possible_ to put in a pre-selected value that the user can never navigate to, etc. I would just argue it is not optimal and can lead to unintended issues, like the following...
![image](https://user-images.githubusercontent.com/14298245/214162969-2f406a7e-4e78-480a-978f-a283f6caa9fb.png)

2. the optgroup is incorrectly indicated as being selected when the select is opened up. The optiongroup can obviously not be selected. It is simply a subcategory title. I believe what's happened here is that the sleight-of-hand being done to hide the pre-populated field value is causing the optgroup to somehow be selected by default.

![image](https://user-images.githubusercontent.com/14298245/214161586-9d7701fe-f2aa-4966-82db-84e8925c7799.png)

3. No option is taking focus when the select list is opened. Such is the expected behaviour, and the failure is again IMO stemming from the decision to populate with a needless 'placeholder' instruction. I have made a small movie to demonstrate how this _should_ be happening

https://user-images.githubusercontent.com/14298245/214163612-4cdc2531-572d-4fe7-b819-e17102c98dbf.mov

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?globals=outline:true&id=components-select--default&viewMode=story

### Steps to reproduce

1. Got to the [storybook for Carbon](https://react.carbondesignsystem.com/iframe.html?globals=outline:true&id=components-select--default&viewMode=story)
2. Select Select from the left nav
3. Open canvas in its own window, by choosing icon in upper right corner 
4. Operate the control by keyboard alone.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
@sstrubberg IMO, these 3 issues should be prioritized as
- issue 1, low sev
- issues 2 and 3 (really 2 manifestations of the same problem), med sev

**Comment by mbgower** [comment]
@thyhmdo @laurenmrice HEre's what an empty default option looks like out of the box, based on modifying https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
![Screenshot 2023-04-27 at 1 49 48 PM](https://user-images.githubusercontent.com/14298245/234988422-f24101bf-beb5-40cb-bf9e-5c11adbb3781.png)
![Screenshot 2023-04-27 at 1 49 56 PM](https://user-images.githubusercontent.com/14298245/234988466-5ad2e1af-03d4-4d93-bf52-14fd7419b06a.png)

**Comment by laurenmrice** [comment]
From the discussion above, for the default example shown in storybook we need to update it to the following:

**Input:**
- Show the input as empty by default.

**List:**
- Do not show categories in the list.
- Have an empty selected option as the first option in the list.

<img width="928" alt="Frame 1" src="https://github.com/carbon-design-system/carbon/assets/43969356/89619848-b3a3-4f88-a1b4-a0c0c08aa5ee">

**Comment by mbgower** [comment]
It's okay to have categories in the list (it's part of the native component) but those should never be choices
